src/CTA.Rules.PortCore/SolutionPort.cs

```diff
@@ -226,8 +226,39 @@ private void InitRules(List<PortCoreConfiguration> solutionConfiguration, List<A
                         }
                     });
                 }
-            }
+                if (projectConfiguration.ProjectType == ProjectType.WCFCodeBasedService || projectConfiguration.ProjectType == ProjectType.WCFConfigBasedService)
+                {
+                    allReferences.Add(Constants.CoreWCFRule);
+                    allReferences.Add(Constants.CoreWCFChannelsRule);
+                    allReferences.Add(Constants.CoreWCFSecurityRule);
+                    allReferences.Add(Constants.CoreWCFDispatcherRule);
+                    allReferences.Add(Constants.CoreWCFDescriptionRule);
+                    allReferences.Add(Constants.CoreWCFDiagnosticsRule);
+                    allReferences.Add(Constants.CoreWCFSecurityTokenRule);
+                    allReferences.Add(Constants.CoreWCFIdentityModelRule);
+                    allReferences.Add(Constants.CoreWCFIdentityModelTokensRule);
+                    allReferences.Add(Constants.CoreWCFIdentityModelClaimsRule);
+                    allReferences.Add(Constants.CoreWCFIdentityModelPolicyRule);
+                    allReferences.Add(Constants.CoreWCFIdentityModelSelectorsRule);
+                    allReferences.Add(Constants.CoreWCFIdentityModelServicesTokensRule);
+                    allReferences.Add(Constants.CoreWCFConfigurationRule);
+                    allReferences.Add(Constants.CoreWCFWebRule);
```

Comment:
is there an API which does AddAll(List<> refrences), would prefer over this.... and initialize the refrences = new List<>{"a","b","c"}().. something like

Reply:
Yes. You can initialize this in a constants somewhere as "WCF" rules. Then use allReference.AddRange(list) to add here.